src/screens/Diagnostics/helpers/createAndShareStateDumpFile.ts

```diff
@@ -0,0 +1,39 @@
+import store from '@/redux/store';
+import { getCircularReferenceReplacer } from '@/screens/Diagnostics/helpers/getCircularReferenceReplacer';
+import RNFS from 'react-native-fs';
+import { APP_STATE_DUMP_FILE_NAME } from '@/screens/Diagnostics/constants';
+import RNShare from 'react-native-share';
+import { IS_ANDROID } from '@/env';
+import { logger, RainbowError } from '@/logger';
+
+export async function createAndShareStateDumpFile() {
+  const appState = store.getState();
+  const stringifiedState = JSON.stringify(
+    appState,
+    getCircularReferenceReplacer()
+  );
```

Comment:
Could just use flatted https://www.npmjs.com/package/flatted

Reply:
I will merge it as is and open another that makes use of this lib, didn't know it existed